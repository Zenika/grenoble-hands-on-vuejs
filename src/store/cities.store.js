import { defineStore } from 'pinia'
import { ref } from 'vue'

export const useCitiesStore = defineStore('cities', () => {
  // State
  const cities = ref([
    {
      name: 'Grenoble',
      latitude: 45.18924,
      longitude: 5.72213
    },
    {
      name: 'Singapour',
      latitude: 1.295600,
      longitude: 103.858995
    },
    {
      name: 'Bordeaux',
      latitude: 44.848089,
      longitude: -0.571017
    },
    {
      name: 'Brest',
      latitude: 48.389397,
      longitude: -4.499237
    },
    {
      name: 'MontrÃ©al',
      latitude: 45.523000,
      longitude: -73.581700
    },
    {
      name: 'Lyon',
      latitude: 45.767443,
      longitude: 4.858798
    },
    {
      name: 'Rennes',
      latitude: 48.113409,
      longitude: -1.661249
    },
    {
      name: 'Nantes',
      latitude: 47.207408,
      longitude: -1.556187
    },
    {
      name: 'Lille',
      latitude: 50.648670,
      longitude: 3.075520
    },
    {
      name: 'Paris',
      latitude: 48.878932,
      longitude: 2.328487
    },
  ])

  // Actions
  const getCityByName = cityName => cities.value.find(city => city.name === cityName)

  return { cities, getCityByName }
})
