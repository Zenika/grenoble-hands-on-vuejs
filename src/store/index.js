import { createStore } from 'vuex'

export default createStore({
  state: {
    latLongMap: {
      GRENOBLE: [45.183916, 5.703630],
      SINGAPOUR: [1.295600, 103.858995],
      BORDEAUX: [44.848089, -0.571017],
      BREST: [48.389397, -4.499237],
      MONTREAL: [45.523000, -73.581700],
      LYON: [45.767443, 4.858798],
      RENNES: [48.113409, -1.661249],
      NANTES: [47.207408, -1.556187],
      LILLE: [50.648670, 3.075520],
      PARIS: [48.878932, 2.328487]
    }
  },
  getters: {
    getMapPosition: state => state.latLongMap,
    getCityPosition: (state, getters) => cityName => getters.getMapPosition[cityName],
    getCities: state => Object.keys(state.latLongMap),
  },
  mutations: {
    createCity(state, city) {
      state.latLongMap[city.name] = [city.latitude, city.longitude]
    }
  },
  actions: {
    createCity(context, city) {
      context.commit('createCity', city)
    }
  },
  modules: {
  }
})
