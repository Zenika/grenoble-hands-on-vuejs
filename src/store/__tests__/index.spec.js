import store, {initialState} from "@/store";

describe('Store', () => {
  it('should match to initialState', () => {
    expect(initialState).toMatchSnapshot()
    expect(store.state).toEqual(initialState)
  });

  it('should contains 3 getters', () => {
    expect(Object.keys(store.getters).length).toBe(3)
  });
  describe('getMapPosition getter', () => {
    it('should return latLongMap', () => {
      expect(store.getters["getMapPosition"]).toEqual(initialState.latLongMap)
    });
  });
  describe('getCityPosition getter', () => {
    it('should be a function that take cityName', () => {
      expect(store.getters["getCityPosition"]).toBeInstanceOf(Function)
    });
    it('should return position of a city', () => {
      expect(store.getters["getCityPosition"]('GRENOBLE')).toEqual([45.183916, 5.703630])
    });
  });
  describe('getCities getter', () => {
    it('should return an array', () => {
      expect(store.getters["getCities"]).toBeInstanceOf(Array)
    });
    it('should return an array with length 10', () => {
      expect(store.getters["getCities"].length).toBe(10)
    });
  });
});
