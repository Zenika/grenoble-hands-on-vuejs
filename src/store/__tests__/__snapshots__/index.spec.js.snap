// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store should match to initialState 1`] = `
Object {
  "latLongMap": Object {
    "BORDEAUX": Array [
      44.848089,
      -0.571017,
    ],
    "BREST": Array [
      48.389397,
      -4.499237,
    ],
    "GRENOBLE": Array [
      45.183916,
      5.70363,
    ],
    "LILLE": Array [
      50.64867,
      3.07552,
    ],
    "LYON": Array [
      45.767443,
      4.858798,
    ],
    "MONTREAL": Array [
      45.523,
      -73.5817,
    ],
    "NANTES": Array [
      47.207408,
      -1.556187,
    ],
    "PARIS": Array [
      48.878932,
      2.328487,
    ],
    "RENNES": Array [
      48.113409,
      -1.661249,
    ],
    "SINGAPOUR": Array [
      1.2956,
      103.858995,
    ],
  },
}
`;
